burn-import/src/burn/node/unary.rs

```diff
@@ -78,122 +90,188 @@ impl<PS: PrecisionSettings> NodeCodegen<PS> for UnaryNode {
 }
 
 impl UnaryNode {
-    pub(crate) fn flatten(
-        input: TensorType,
-        output: TensorType,
-        start_dim: usize,
-        end_dim: usize,
-    ) -> Self {
+    pub(crate) fn flatten(input: Type, output: Type, start_dim: usize, end_dim: usize) -> Self {
         let start_dim = start_dim.to_tokens();
         let end_dim = end_dim.to_tokens();
         let function = move |input| quote! { #input.flatten(#start_dim, #end_dim) };
 
         Self::new(input, output, UnaryNodeKind::Flatten, Arc::new(function))
     }
 
-    pub(crate) fn relu(input: TensorType, output: TensorType) -> Self {
+    pub(crate) fn relu(input: Type, output: Type) -> Self {
         let function = move |input| quote! { burn::tensor::activation::relu(#input) };
         Self::new(input, output, UnaryNodeKind::Relu, Arc::new(function))
     }
 
-    pub(crate) fn sigmoid(input: TensorType, output: TensorType) -> Self {
+    pub(crate) fn sigmoid(input: Type, output: Type) -> Self {
         let function = move |input| quote! { burn::tensor::activation::sigmoid(#input) };
         Self::new(input, output, UnaryNodeKind::Sigmoid, Arc::new(function))
     }
 
-    pub(crate) fn log_softmax(input: TensorType, output: TensorType, dim: usize) -> Self {
+    pub(crate) fn log_softmax(input: Type, output: Type, dim: usize) -> Self {
         let dim = dim.to_tokens();
         let function = move |input| quote! { burn::tensor::activation::log_softmax(#input, #dim) };
         Self::new(input, output, UnaryNodeKind::LogSoftmax, Arc::new(function))
     }
 
-    pub(crate) fn transpose(input: TensorType, output: TensorType) -> Self {
+    pub(crate) fn transpose(input: Type, output: Type) -> Self {
         let function = move |input| quote! { #input.transpose() };
         Self::new(input, output, UnaryNodeKind::Transpose, Arc::new(function))
     }
+
+    pub(crate) fn cast(input: Type, output: Type) -> Self {
+        let function = match output.clone() {
+            Type::Scalar(scalar) => {
+                let ty = scalar.ty();
+                move |input| quote! { #input as #ty }
```

Comment:
If the output type is a tensor, we should generate another code block.

```rust
Tensor::from_data(Data::from([#input]).convert()).unsqueeze();
```

Reply:
Got it. I have added a todo for supporting to cast other types.